### PR TITLE
PLF-8642:Wrong digest mail (Daily / Weekly) are sent to users concern…

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**
In the digest mail, the problem was to send mail for all users without specify who is the receiver. So I add a condition to specify who is allowed to receive mail.

### CODE REVIEW

File: integration/src/main/java/org/exoplatform/task/integration/notification/AbstractNotificationPlugin.java [hunk 1]
```diff
@@ -76,6 +79,10 @@ public NotificationInfo makeNotification(NotificationContext ctx) {
                                .with(NotificationUtils.ACTIVITY_ID, activityId.toString())
                                .with(NotificationUtils.TASK_URL, taskUrl)
                                .with(NotificationUtils.PROJECT_URL, projectUrl)
+                               .with(NotificationUtils.Assign_Task, assignTask)
+                               .with(NotificationUtils.ListOfCoworker, String.valueOf(listOfCowoker))
+                               .with(NotificationUtils.Coworker_Task, String.valueOf(coworker))
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/AbstractNotificationPlugin.java]:
```suggestion
                               .with(NotificationUtils.ADDED_COWORKER, String.valueOf(coworker))
```

File: integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java [hunk 2]
```diff
@@ -88,8 +88,11 @@ protected MessageInfo makeMessage(NotificationContext ctx) {
       String commentText = notification.getValueOwnerParameter(NotificationUtils.COMMENT_TEXT);
       String taskUrl = notification.getValueOwnerParameter(NotificationUtils.TASK_URL);
       String projectUrl = notification.getValueOwnerParameter(NotificationUtils.PROJECT_URL);
-
+      String assignTask = notification.getValueOwnerParameter(NotificationUtils.Assign_Task);
+      String ListOfCoworker = notification.getValueOwnerParameter(NotificationUtils.ListOfCoworker);
       commentText = CommentUtil.formatMention(getExcerpt(commentText, 130), userService);
+      String coworker = notification.getValueOwnerParameter(NotificationUtils.Coworker_Task);
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java]:
```suggestion
      String coworker = notification.getValueOwnerParameter(NotificationUtils.ADDED_COWORKER);
```

File: integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java [hunk 3]
```diff
@@ -88,8 +88,11 @@ protected MessageInfo makeMessage(NotificationContext ctx) {
...
+      String usersMentioned = notification.getValueOwnerParameter(NotificationUtils.UsersMENTIONED);
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java]:
```suggestion
      String usersMentioned = notification.getValueOwnerParameter(NotificationUtils.MENTIONED_USERS);
```

File: integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java [hunk 4]
```diff
@@ -111,6 +114,10 @@ protected MessageInfo makeMessage(NotificationContext ctx) {
       templateContext.put("DUE_DATE", getDueDate(notification));
       templateContext.put("TASK_URL", taskUrl);
       templateContext.put("PROJECT_URL", projectUrl);
+      templateContext.put("AssignTask", assignTask);
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java]:
```suggestion
      templateContext.put("AssignTask", assigne);
```

File: integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java [hunk 5]
```diff
@@ -111,6 +114,10 @@ protected MessageInfo makeMessage(NotificationContext ctx) {
...
+      templateContext.put("listOfCoworker", ListOfCoworker);
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java]:
```suggestion
      templateContext.put("listOfCoworker", listOfCoworker);
```

File: integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java [hunk 6]
```diff
@@ -128,6 +135,18 @@ protected boolean makeDigest(NotificationContext ctx, Writer writer) {
 
       List<NotificationInfo> notifications = ctx.getNotificationInfos();
       NotificationInfo first = notifications.get(0);
+      if (TaskAssignPlugin.ID.equals(first.getKey().getId()) && !first.getTo().equals(first.getOwnerParameter().get(NotificationUtils.Assign_Task)) && !first.getOwnerParameter().get(NotificationUtils.CREATOR.getKey()).equals(first.getOwnerParameter().get(NotificationUtils.Assign_Task))) {
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/MailTemplateProvider.java]: extract function calls like `first.getOwnerParameter().get(NotificationUtils.CREATOR.getKey())`  and `first.getKey().getId()` into variables to avoid calling them in each condition check

File: integration/src/main/java/org/exoplatform/task/integration/notification/NotificationUtils.java [hunk 7]
```diff
@@ -42,5 +42,9 @@
   public final static String ACTIVITY_ID = "activityId";
   public static final String TASK_URL = "taskUrl";
   public static final String PROJECT_URL = "projectUrl";
+  public static final String Assign_Task = "assignTask";
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/NotificationUtils.java]: Update the constants with suggested values above

File: integration/src/test/java/org/exoplatform/task/integration/notification/TemplateUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package org.exoplatform.task.integration.notification;
```
Comment (ahamdi) [inline integration/src/test/java/org/exoplatform/task/integration/notification/TemplateUtilsTest.java]: This test is not related to the fixed bug.

**Comment by Rdidmohamedelhadi** [comment]
> makeDigest

**Comment by Rdidmohamedelhadi** [comment]
> Please add Unit Test that covers all the cases in MailTemplateProvider.makeDigest()

ok
